imports/plugins/included/product-variant/containers/productGridContainer.js

```diff
@@ -0,0 +1,165 @@
+      initialLoad: true,
+      slug: "",
+      canLoadMoreProducts: false
+    };
+  }
+
+  componentWillMount() {
+    const selectedProducts = Reaction.getUserPreferences("reaction-product-variant", "selectedGridItems");
+    const products = this.products;
+
+    if (_.isEmpty(selectedProducts)) {
+      return Reaction.hideActionView();
+    }
+
+    // Save the selected items to the Session
+    Session.set("productGrid/selectedProducts", _.uniq(selectedProducts));
+
+    if (products) {
+      const filteredProducts = _.filter(products, (product) => {
+        return _.includes(selectedProducts, product._id);
+      });
+
+      if (Reaction.isPreview() === false) {
+        Reaction.showActionView({
+          label: "Grid Settings",
+          i18nKeyLabel: "gridSettingsPanel.title",
+          template: "productSettings",
+          type: "product",
+          data: { products: filteredProducts }
+        });
+      }
+    }
+  }
+
+  componentWillReceiveProps = (nextProps) => {
+    this.setState({
+      products: nextProps.products,
+      productIds: nextProps.productIds,
+      productsByKey: nextProps.productsByKey
+    });
+  }
+
+  handleSelectProductItem = (isChecked, productId) => {
+    let selectedProducts = Session.get("productGrid/selectedProducts");
+
+    if (isChecked) {
+      selectedProducts.push(productId);
+    } else {
+      selectedProducts = _.without(selectedProducts, productId);
+    }
+
+    Reaction.setUserPreferences("reaction-product-variant", "selectedGridItems", selectedProducts);
+
+    // Save the selected items to the Session
+    Session.set("productGrid/selectedProducts", _.uniq(selectedProducts));
+
+    const products = this.products;
+
+    if (products) {
+      const filteredProducts = _.filter(products, (product) => {
+        return _.includes(selectedProducts, product._id);
+      });
+
+      Reaction.showActionView({
+        label: "Grid Settings",
+        i18nKeyLabel: "gridSettingsPanel.title",
+        template: "productSettings",
+        type: "product",
+        data: { products: filteredProducts }
+      });
+    }
+  }
+
+  handleProductDrag = (dragIndex, hoverIndex) => {
+    const newState = this.changeProductOrderOnState(dragIndex, hoverIndex);
+    this.setState(newState, this.callUpdateMethod);
+  }
+
+  changeProductOrderOnState(dragIndex, hoverIndex) {
+    const product = this.state.productIds[dragIndex];
+
+    return update(this.state, {
+      productIds: {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, product]
+        ]
+      }
+    });
+  }
+
+  callUpdateMethod() {
+    const tag = ReactionProduct.getTag();
+
+    this.state.productIds.map((productId, index) => {
+      const position = { position: index, updatedAt: new Date() };
+
+      Meteor.call("products/updateProductPosition", productId, position, tag, error => {
+        if (error) {
+          Logger.warn(error);
+          throw new Meteor.Error(403, error);
```

Comment:
We want to get away from throwing these numeric errors, especially since 403 should always mean "Access Denied" and you don't know if that's the issue here.